blocks/shuffle/src/shuffle.tsx

```diff
@@ -186,40 +164,42 @@ export const Shuffle: BlockComponent<BlockEntityProperties> = ({
     updateItems([], true);
   };
 
-  // adds the items selected on AddEntitiesDialog to the list
-  const handleAddEntityItems = (entityItems: Items) => {
-    updateItems(draftItems.concat(entityItems), true);
-  };
-
   /**
    * linked items does not store a `value`,
    * instead we use `entityId` to set the up-to-date entity label as `value`
    * this way, we don't show a stale `value` if the linked entity gets updated
    */
   const enhancedDraftItems = useMemo(() => {
-    // creating maps here, so we can use maps in the loop below instead of Array.find
-    // avoid using nested loops for better performance
-    const linkedEntitiesMap = new Map(
-      blockGraph?.linkedEntities.map((entity) => [entity.entityId, entity]),
+    const outgoingLinks = getOutgoingLinksForEntity(
+      blockEntitySubgraph,
+      rootEntity.metadata.recordId.entityId,
     );
 
-    const entityTypesMap = new Map(
-      entityTypes.map((type) => [type.entityTypeId, type]),
+    const linksMap = new Map(
+      outgoingLinks.map((link) => [
+        link.metadata.recordId.entityId,
+        getRightEntityForLinkEntity(
+          blockEntitySubgraph,
+          link.metadata.recordId.entityId,
+        ),
+      ]),
     );
 
     return draftItems.map((item) => {
```

Comment:
Have we removed the ability to add items that represent entities at the moment due to the missing `queryEntityTypes` message?

Reply:
Yes – you can pass in data that represents that, but you can't load it in via UI. Was agreed on slack somewhere.